### PR TITLE
Add support for workload_autoscaler_profile settings

### PR DESCRIPTION
**Pull request by bonddim**
## Describe your changes
Add support for `workload_autoscaler_profile` settings by adding variables:
`workload_autoscaler_profile_keda_enabled` - to enable Keda Autoscaler

## Checklist before requesting a review
- [x] The pr title can be used to describe what this pr did in `CHANGELOG.md` file
- [x] I have executed pre-commit on my machine
- [x] I have passed pr-check on my machine

### CODE REVIEW

**Comment by lonegunmanb** [comment]
Thanks @bonddim for opening this pr! I've checked the [document](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#keda_enabled), the feature seems to be a preview one, I'd like to hold this pr until the feature is GA. Thanks for your contribution and please feel free to ping me once the feature is GA.

**Comment by samuelb** [comment]
KEDA is now GA: https://azure.microsoft.com/en-us/updates/ga-kubernetes-eventdriven-autoscaling-keda-addon-for-aks/

**Comment by bonddim** [comment]
> KEDA is now GA: https://azure.microsoft.com/en-us/updates/ga-kubernetes-eventdriven-autoscaling-keda-addon-for-aks/

Thank you for notifying me about the update!

**Comment by bonddim** [comment]
> Thanks @bonddim for opening this pr! I've checked the [document](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#keda_enabled), the feature seems to be a preview one, I'd like to hold this pr until the feature is GA. Thanks for your contribution and please feel free to ping me once the feature is GA.

@lonegunmanb, PR is updated with latest main and removed VPA variable.

**Comment by lonegunmanb** [comment]
> > Thanks @bonddim for opening this pr! I've checked the [document](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#keda_enabled), the feature seems to be a preview one, I'd like to hold this pr until the feature is GA. Thanks for your contribution and please feel free to ping me once the feature is GA.
> 
> @lonegunmanb, PR is updated with latest main and removed VPA variable.

Wow thanks for the update @bonddim @samuelb !

File: variables.tf [hunk 1]
```diff
@@ -1285,6 +1285,12 @@ variable "web_app_routing" {
 EOT
 }
 
+variable "workload_autoscaler_profile_keda_enabled" {
```
Comment (lonegunmanb) [inline variables.tf:1288]: Could we change this variable to the following?:

```hcl
variable "workload_autoscaler_profile" {
  type = object({
    keda_enabled                    = optional(bool, false)
    vertical_pod_autoscaler_enabled = optional(bool, false)
  })
  default     = null
  description = <<-EOT
 - `keda_enabled` - (Optional) Specifies whether KEDA Autoscaler can be used for workloads.
 - `vertical_pod_autoscaler_enabled` - (Optional) Specifies whether Vertical Pod Autoscaler should be enabled.
EOT
}
```

File: main.tf [hunk 2]
```diff
@@ -493,6 +493,9 @@ resource "azurerm_kubernetes_cluster" "main" {
       dns_zone_id = var.web_app_routing.dns_zone_id
     }
   }
+  workload_autoscaler_profile {
```
Comment (lonegunmanb) [inline main.tf:496]: Could we use `dynamic` block here? Like:

```hcl
dynamic "workload_autoscaler_profile" {
    for_each = var.workload_autoscaler_profile == null ? [] : [var.workload_autoscaler_profile]
    content {
      keda_enabled                    = workload_autoscaler_profile.value.keda_enabled
      vertical_pod_autoscaler_enabled = workload_autoscaler_profile.value.vertical_pod_autoscaler_enabled
    }
  }
```

**Review by lonegunmanb — CHANGES_REQUESTED** [review]
Thanks @bonddim for the update.

According to the [provider's schema](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/containers/kubernetes_cluster_resource.go#L1492-L1507), there're two configurable attributes in `workload_autoscaler_profile` block, I'd like to give this nested block a full support by this pr, would you please adjust the pr as I'v commented? Thanks!

**Comment by bonddim** [comment]
> Thanks @bonddim for the update.
> 
> According to the [provider's schema](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/containers/kubernetes_cluster_resource.go#L1492-L1507), there're two configurable attributes in `workload_autoscaler_profile` block, I'd like to give this nested block a full support by this pr, would you please adjust the pr as I'v commented? Thanks!

I removed VPA setting, because it's still in preview.

File: main.tf (hunk 2 above)
Comment (bonddim) [reply main.tf]: Done

**Review by lonegunmanb — APPROVED** [review]
Thanks @bonddim, LGTM!

**Pull request merged**